bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/ReplicationWorker.java

```diff
@@ -271,22 +265,20 @@ private boolean rereplicate(long ledgerIdToReplicate) throws InterruptedExceptio
             underreplicationManager.markLedgerReplicated(ledgerIdToReplicate);
             getExceptionCounter("BKNoSuchLedgerExistsException").inc();
             return false;
-        } catch (BKReadException e) {
-            LOG.info("BKReadException while"
-                    + " opening ledger {} for replication."
-                    + " Enough Bookies might not have available"
-                    + " So, no harm to continue", ledgerIdToReplicate);
-            underreplicationManager
-                    .releaseUnderreplicatedLedger(ledgerIdToReplicate);
-            getExceptionCounter("BKReadException").inc();
-            return false;
-        } catch (BKBookieHandleNotAvailableException e) {
-            LOG.info("BKBookieHandleNotAvailableException while"
-                    + " opening ledger {} for replication."
+        } catch (BKException e) {
+            LOG.info("{} while"
+                    + " rereplicating ledger {}."
                     + " Enough Bookies might not have available"
-                    + " So, no harm to continue", ledgerIdToReplicate);
+                    + " So, no harm to continue",
+                e.getClass().getSimpleName(),
+                ledgerIdToReplicate);
             underreplicationManager
                     .releaseUnderreplicatedLedger(ledgerIdToReplicate);
+            getExceptionCounter(e.getClass().getSimpleName()).inc();
+
+            if (e instanceof BKNotEnoughBookiesException) {
```

Comment:
Not a big fan of this. Maybe create a logErrorAndRelease(BKException e, long ledgerId) method to move out the common stuff and handle BKNotEnoughBookiesException in a different catch clause.